server/text_generation_server/utils/weights.py

```diff
@@ -110,21 +111,92 @@ def get_weights_col_packed(
         prefix: str,
         block_sizes: Union[int, List[int]],
     ):
-
-        return self.weight_class(
-            weights.get_packed_sharded(
-                f"{prefix}.weight", dim=0, block_sizes=block_sizes
-            ),
+        w = weights.get_packed_sharded(
+            f"{prefix}.weight", dim=0, block_sizes=block_sizes
         )
 
+        if w.dtype == torch.float8_e4m3fn:
+            # FIXME: here to avoid circular import
+            from text_generation_server.layers.fp8 import Fp8Weight
```

Comment:
Not happy about this.

Reply:
Hmmm, this breaks the abstraction quite a bit. This class is also used by other (explicit) quantizers like eetq.

I think it would make more sense to put this implementation in something like a `HybridFP8FP16Loader` in the `fp8` module. Then we could add some logic to the [`get_loader`](https://github.com/huggingface/text-generation-inference/blob/e52be9bba2a11f6b2145fcd21be0ba860b977a4d/server/text_generation_server/utils/quantization.py#L95) function, along the lines of: when `quantizer==None` is set, enumerate over the tensors (should be cheap I think for just getting the dtypes?) and then if an FP8 weight is encountered,

Reply:
Agreed, modified the code to add another loader.